admin/admin_page_registration.php

```diff
@@ -5,91 +5,88 @@
  * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
  * @version $Id: DrByte 2020 Jun 07 Modified in v1.5.7 $
  */
-require('includes/application_top.php');
+require 'includes/application_top.php';
 
 // initialise form values
 $page_key = $language_key = $main_page = $page_params = $menu_key = $checked = '';
 $sort_order = 0;
 
 // check if we are receiving form content and if so validate and process it
 if (isset($_POST) && !empty($_POST)) {
-  $error = FALSE;
+    $error = FALSE;
 
-  if (isset($_POST['page_key'])) {
-    $page_key = zen_db_prepare_input($_POST['page_key']);
-  }
-  if (empty($page_key)) {
-    $error = TRUE;
-    $messageStack->add(ERROR_PAGE_KEY_NOT_ENTERED, 'error');
-  } elseif (zen_page_key_exists($page_key)) {
-    $error = TRUE;
-    $messageStack->add(ERROR_PAGE_KEY_ALREADY_EXISTS, 'error');
-  }
+    $keys = [
+        'page_key' => [
+            'emptyChk' => true,
+            'elseifFunc' => 'zen_page_key_exists',
+        ],
+        'language_key' => [
+            'emptyChk' => true,
+            'elseifFunc' => 'defined',
+            'elseifNot' => true,
+        ],
+        'main_page' => [
+            'emptyChk' => true,
+            'elseifFunc' => 'defined',
+            'elseifNot' => true,
+        ],
+        'page_params',
+        'menu_key' => [
+            'emptyChk' => true,
+        ],
+        'sort_order' => [
+            'valueType' => 'integer',
+        ],
+    ];
 
-  if (isset($_POST['language_key'])) {
-    $language_key = zen_db_prepare_input($_POST['language_key']);
-  }
-  if (empty($language_key)) {
-    $error = TRUE;
-    $messageStack->add(ERROR_LANGUAGE_KEY_NOT_ENTERED, 'error');
-  } elseif (!defined($language_key)) {
-    $error = TRUE;
-    $messageStack->add(ERROR_LANGUAGE_KEY_HAS_NOT_BEEN_DEFINED, 'error');
-  }
+    foreach ($keys as $key => $value) {
+        if (isset($_POST[$key])) {
+            ${$key} = $_POST[$key];
+            if (!empty($value['valueType'])) {
+                ${$key} = $db->bindVars(':val:', ':val:', ${$key}, $value['valueType']);
+            }
+            ${$key} = zen_db_prepare_input(${$key});
+        }
+        if (!empty($value['emptyChk'])) {
+            if (empty(${$key})) {
+                $error = TRUE;
+                $messageStack->add(constant('ERROR_' . strtoupper($key) . '_NOT_ENTERED'), 'error');
+            } else if (!empty($value['elseifFunc']) && ($value['elseifFunc'](${$key})) ? /* result was true */ empty($value['elseifNot']) : /* result was false */ !empty($value['elseifNot'])) {
+                $error = TRUE;
+                $message = 'ERROR_' . strtoupper($key) . '_ALREADY_EXISTS';
+                if ($value['elseifFunc'] == 'defined' && !empty($value['elseifNot'])) {

```

Comment:
Almost seems that for proper logic that a separate key should be used here instead of being dependent on `elseifNot` being set... Yes, this was "born" of the original data; however, currently there are two types of test here, that there is either a duplicate or an absence of information... Thing is though, that the current logic of if it is to check that it is `not defined` (perhaps should swap the two tests for a more "logical" sequence) that then these dependencies are correct and an additional key would only make things easier to muck up.